pageserver/src/tenant.rs

```diff
@@ -2212,6 +2213,39 @@ impl Tenant {
 
         Ok(())
     }
+
+    pub(crate) fn get_sizes(&self) -> TopNTenantShardItem {
+        let mut result = TopNTenantShardItem {
+            id: self.tenant_shard_id,
+            resident_size: 0,
+            physical_size: 0,
+            max_logical_size: 0,
+        };
+
+        let timelines = self
+            .timelines
+            .lock()
+            .unwrap()
+            .values()
+            .cloned()
+            .collect::<Vec<_>>();
+
+        for timeline in timelines {
```

Comment:
I suppose this could use the `Iterator::fold` function and work on the timelines iterator directly instead of first collecting to a vec and then iterating over it.

Reply:
The choice here is between the cost of holding the timelines lock while we do all the size reads, or the cost of cloning timeline Arcs.  Now that the size read is no longer taking a mutex, it's probably fine to hold the timelines mutex across the size reads (24d137a52)